core-bundle/src/Image/Studio/FigureBuilder.php

```diff
@@ -290,7 +295,14 @@ public function fromUrl(string $url, array $baseUrls = []): self
      */
     public function fromImage(ImageInterface $image): self
     {
-        return $this->fromPath($image->getPath());
+        $this->image = $image;
+        $this->filePath = $image->getPath();
```

Comment:
Couldn't we keep the existing logic and just add the line above? Then the exception handling would also already work.